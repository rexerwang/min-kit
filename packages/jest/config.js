/**
 * @typedef {import('jest').Config Config}
 */

const defineJestConfig = require('./config/taro-jest/index').default

/** @type {Config} */
const baseConfig = defineJestConfig({
  globalSetup: require.resolve('./config/globalSetup.js'),
  setupFilesAfterEnv: [require.resolve('./config/setupAfterEnv.js')],
  testEnvironment: '@happy-dom/jest-environment', // 'jsdom'
  testMatch: ['<rootDir>/src/__tests__/**/*.test.ts?(x)'],
  collectCoverageFrom: ['<rootDir>/src/**/*.ts?(x)', '!**/__tests__/**'],
  coverageReporters: process.env.CI ? ['json'] : ['html', 'text'],
  passWithNoTests: true,
})

/**
 * defineJestConfig
 * @param {Partial<Config>} [config]
 * @returns {Config}
 */
module.exports = (config = {}) => {
  let { setupFilesAfterEnv, collectCoverageFrom, ...configs } = baseConfig

  if (Array.isArray(config.setupFilesAfterEnv)) {
    setupFilesAfterEnv = setupFilesAfterEnv.concat(config.setupFilesAfterEnv)
  }

  if (Array.isArray(config.collectCoverageFrom)) {
    collectCoverageFrom = [collectCoverageFrom[0], ...config.collectCoverageFrom, collectCoverageFrom[1]]
  }

  return {
    ...configs,
    ...config,
    setupFilesAfterEnv,
    collectCoverageFrom,
  }
}
