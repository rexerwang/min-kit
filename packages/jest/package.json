{
  "name": "@min-kit/jest",
  "version": "0.5.11",
  "private": true,
  "description": "jest config & utils for internal",
  "author": "rexerwang",
  "license": "MIT",
  "repository": "rexerwang/min-kit",
  "exports": {
    ".": "./utils.ts",
    "./config": "./config.js"
  },
  "devDependencies": {
    "@happy-dom/jest-environment": "^14.7.1",
    "@tarojs/test-utils-react": "^0.1.1",
    "@testing-library/jest-dom": "^6.4.8",
    "@testing-library/react": "^16.0.1"
  }
}
