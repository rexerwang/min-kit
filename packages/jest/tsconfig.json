{
  "extends": "../../tsconfig.root.json",
  "include": ["utils.ts"]
}
