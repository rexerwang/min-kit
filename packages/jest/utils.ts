import TestUtils from '@tarojs/test-utils-react'

export const createTaroTestUtils = () => {
  const testUtils = new TestUtils()
  return Object.assign(testUtils, { unmount: testUtils.unmout })
}

export const spyOnConsole = (methods = ['error', 'warn', 'info', 'debug']) =>
  methods.map((method: any) => {
    const spy = jest.spyOn(console, method)
    spy.mockImplementation(jest.fn())
    return spy
  })

export * from '@testing-library/react'
