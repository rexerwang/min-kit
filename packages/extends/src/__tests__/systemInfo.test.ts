import { spyOnConsole } from '@min-kit/jest'
import Taro from '@tarojs/taro'

describe('systemInfo', () => {
  spyOnConsole('error') // disable error

  it('should SystemInfo toMatchSnapshot when getSystemInfo successfully', async () => {
    const { SystemInfo } = await import('../systemInfo')
    expect(SystemInfo).toMatchSnapshot()
  })

  it('should the default SystemInfo toMatchSnapshot when getSystemInfo failed', async () => {
    jest.spyOn(Taro, 'getSystemInfoSync').mockImplementation(jest.fn().mockReturnValue(null))
    jest.spyOn(Taro, 'getMenuButtonBoundingClientRect').mockImplementation(jest.fn().mockReturnValue(null))

    await jest.isolateModulesAsync(async () => {
      const { SystemInfo } = await import('../systemInfo')
      expect(SystemInfo).toMatchSnapshot()
    })
  })
})
