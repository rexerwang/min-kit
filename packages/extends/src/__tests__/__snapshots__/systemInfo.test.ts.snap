// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`systemInfo should SystemInfo toMatchSnapshot when getSystemInfo successfully 1`] = `
{
  "SDKVersion": "3.1.2",
  "albumAuthorized": true,
  "batteryLevel": 76,
  "benchmarkLevel": -1,
  "bluetoothAuthorized": true,
  "bluetoothEnabled": true,
  "brand": "devtools",
  "cameraAuthorized": false,
  "deviceOrientation": "portrait",
  "devicePixelRatio": 3,
  "enableDebug": false,
  "fontSizeScaleFactor": 1,
  "fontSizeSetting": 16,
  "host": {
    "env": "WeChat",
  },
  "isAndroid": false,
  "isDevtools": true,
  "isiOS": true,
  "isiPad": false,
  "isiPhoneX": true,
  "language": "zh_CN",
  "locationAuthorized": false,
  "locationEnabled": true,
  "locationReducedAccuracy": true,
  "memorySize": 2048,
  "menuButtonRect": {
    "bottom": 80,
    "height": 32,
    "left": 281,
    "right": 368,
    "top": 48,
    "width": 87,
  },
  "microphoneAuthorized": true,
  "mode": "default",
  "model": "iPhone X",
  "notificationAlertAuthorized": true,
  "notificationAuthorized": true,
  "notificationBadgeAuthorized": true,
  "notificationSoundAuthorized": true,
  "phoneCalendarAuthorized": false,
  "pixelRatio": 3,
  "platform": "devtools",
  "safeArea": {
    "bottom": 778,
    "height": 734,
    "left": 0,
    "right": 375,
    "top": 44,
    "width": 375,
  },
  "screenHeight": 812,
  "screenTop": 88,
  "screenWidth": 375,
  "statusBarHeight": 44,
  "system": "iOS 10.0.1",
  "version": "8.0.5",
  "wifiEnabled": true,
  "windowHeight": 724,
  "windowWidth": 375,
}
`;

exports[`systemInfo should the default SystemInfo toMatchSnapshot when getSystemInfo failed 1`] = `
{
  "SDKVersion": "",
  "benchmarkLevel": 0,
  "brand": "",
  "isAndroid": false,
  "isDevtools": false,
  "isiOS": false,
  "isiPad": false,
  "isiPhoneX": false,
  "language": "",
  "menuButtonRect": {
    "bottom": 0,
    "height": 0,
    "left": 0,
    "right": 0,
    "top": 0,
    "width": 0,
  },
  "model": "",
  "pixelRatio": 0,
  "platform": "",
  "safeArea": {
    "bottom": 0,
    "height": 0,
    "left": 0,
    "right": 0,
    "top": 0,
    "width": 0,
  },
  "screenHeight": 0,
  "screenWidth": 0,
  "statusBarHeight": 0,
  "system": "",
  "version": "",
  "windowHeight": 0,
  "windowWidth": 0,
}
`;
